### PR TITLE
Fix race condition(s) while adding/removing a light/device

### PR DESCRIPTION
**Pull request by marcelveldt**
Fix some race conditions when adding/removing a light/device resulting in a (temporary) KeyError.

fixes https://github.com/home-assistant/core/issues/63591

### CODE REVIEW

File: aiohue/v2/controllers/devices.py [hunk 1]
```diff
@@ -19,17 +19,11 @@ class DevicesController(BaseResourcesController[Type[Device]]):
 
     def get_lights(self, id: str) -> List[Light]:
         """Return all lights belonging to given device."""
-        return [
-            self._bridge.lights[x] for x in self[id].lights if x in self._bridge.lights
-        ]
+        return [x for x in self._bridge.lights if x.id in self[id].lights]
```
Comment (balloob) [inline aiohue/v2/controllers/devices.py:22]: you're only returning `x` here, which I think is the key. However it's typed as returning the light object. Is this a breaking change?

Comment (marcelveldt) [reply aiohue/v2/controllers/devices.py:22]: Good catch, that is actually a typo. I'll fix it

Comment (marcelveldt) [reply aiohue/v2/controllers/devices.py:22]: Actually the code is fine and the actual light object is returned here so all good...

Schermafbeelding 2022-02-24 om 20 41 15

Comment (marcelveldt) [reply aiohue/v2/controllers/devices.py:22]: tested the code in HA too to see if it fixes the reported issue when adding a new light and it's all fine now

**Review by balloob — APPROVED** [review]

**Pull request merged**